lib/bats-core/common.bash

```diff
@@ -197,22 +197,23 @@ bats_trim() {                                            # <output-variable> <st
 # a helper function to work around unbound variable errors with ${arr[@]} on Bash 3
 bats_append_arrays_as_args() { # <array...> -- <command ...>
   local -a trailing_args=()
-  while (($# > 0)) && [[ $1 != -- ]]; do
-    local array=$1
+  local array
+  while (( $# > 0 )) && [[ $1 != -- ]]; do
+    array=$1
     shift
 
-    if eval "(( \${#${array}[@]} > 0 ))"; then
-      eval "trailing_args+=(\"\${${array}[@]}\")"
+    if (( ${#array[@]} > 0 )); then
```

Comment:
This totally changes the meaning. I think the documentation of this function could be improved but the tests and usage should make it clear:

```
array_1=(a b c)
array_2=(1 2)
bats_append_arrays_as_args array_1 array_2 -- echo
````
will ultimately call
```
echo a b c 1 2
```
so it is a bit like `xargs` but instead of consuming everything from stdin, the arguments before `--` are arrays whose elements should be appended as parameters to the command.